easybuild/easyblocks/p/python.py

```diff
@@ -128,6 +132,39 @@ def __init__(self, *args, **kwargs):
             easybuild_subdir = log_path()
             self.pythonpath = os.path.join(easybuild_subdir, 'python')
 
+        ext_defaults = {
+            # The only installed packages at this point are default installed ones, e.g. pip&setuptools
+            # And we want to upgrade them cleanly, i.e. uninstall them
+            'pip_ignore_installed': False,
```

Comment:
The intention here is to avoid that `--ignore-installed` is used when updating `pip` and `setuptools`, but you're effectively disabling it for *all* extensions being installed along with Python (unless otherwise specified explicitly in the easyconfig file via `exts_default_options`)?

Reply:
Enhanced the comment as suggested. 
I found another reason for that: Similar for pip+setuptools we MUST actually use this for any other extension that we install twice (for whatever reason) or we will have those Frankenstein-software again with half-updated files.
This could happen by accident, e.g. auto-installing a dependency of an extension and then the same package as an extension, or intentionally by e.g. installing an older version first and upgrading it later (e.g. when other packages were installed which are required for the newer version)

Reply:
OK, thanks for clarifying the comment!